state-commitments/optimized/inclusion_proof.go

```diff
@@ -0,0 +1,113 @@
+package optimized
+
+import (
+	"errors"
+	"math"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Computes the Merkle proof for a leaf at a given index.
+// It uses the last leaf to pad the tree up to the 'virtual' size if needed.
+func (h *HistoryCommitter) computeMerkleProof(leafIndex uint64, leaves []common.Hash, virtual uint64) ([]common.Hash, error) {
+	if len(leaves) == 0 {
+		return nil, nil
+	}
+	ok, err := isPowTwo(virtual)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, errors.New("virtual size must be a power of 2")
+	}
+	if leafIndex >= uint64(len(leaves)) {
+		return nil, errors.New("leaf index out of bounds")
+	}
+	if virtual < uint64(len(leaves)) {
+		return nil, errors.New("virtual size must be greater than or equal to the number of leaves")
+	}
+	numRealLeaves := uint64(len(leaves))
+	lastLeaf, err := h.hash(leaves[numRealLeaves-1][:])
+	if err != nil {
+		return nil, err
+	}
+	depth := int(math.Ceil(math.Log2(float64(virtual))))
```

Comment:
I think this should be using the new fast Ilog2.

Reply:
Done.